libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/sound/MapboxAudioViewModel.kt

```diff
@@ -0,0 +1,48 @@
+package com.mapbox.navigation.dropin.component.sound
+
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.dropin.lifecycle.UIViewModel
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.emptyFlow
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.launch
+
+sealed class AudioAction {
+    object Mute : AudioAction()
+    object Unmute : AudioAction()
+    object Toggle : AudioAction()
+}
+
+@ExperimentalPreviewMapboxNavigationAPI
+class MapboxAudioViewModel : UIViewModel<MapboxAudioState, AudioAction>(MapboxAudioState()) {
```

Comment:
No unit tests?

Reply:
Yeah, always good to have unit test definitions. Have been meaning to.

Added some for `UIViewModel` too. While migrating, we also noticed things to improve